### ISSUE TITLE
Clippy suggests incorrect fix

### ISSUE THREAD

**Issue by he-thinks** [rust-lang/rust-clippy#11732]
### Summary

Clippy suggests incorrect fixes for `#[warn(unused_parens)]` for these patterns (the same issue with the stable 1.73.0 rustc):
```rs
let(mut _x): i8 = 0;
let( mut _x): i8 = 0;
let(_x): i8 = 0;
```

### Reproducer

I tried this code:

```rust
fn main() {
    let(mut _x): i8 = 0;
}
```

I expected to see this happen:
```bash
warning: unnecessary parentheses around pattern
 --> src/main.rs:2:8
  |
2 |     let(mut _x): i8 = 0;
  |        ^      ^
  |
  = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
  |
2 -     let(mut _x): i8 = 0;
2 +     let mut _x: i8 = 0;
  |
```

Instead, this happened:
```bash
warning: unnecessary parentheses around pattern
 --> src/main.rs:2:8
  |
2 |     let(mut _x): i8 = 0;
  |        ^      ^
  |
  = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
  |
2 -     let(mut _x): i8 = 0;
2 +     letmut _x: i8 = 0;
  |
```

### Version

```text
rustc 1.75.0-nightly (e5cfc5547 2023-10-28)
binary: rustc
commit-hash: e5cfc55477eceed1317a02189fdf77a4a98f2124
commit-date: 2023-10-28
host: x86_64-pc-windows-gnu
release: 1.75.0-nightly
LLVM version: 17.0.3
```

**Comment by y21** [comment]
Shouldn't this be on the rust-lang/rust repo? `unused_parens` is a rustc lint, not a clippy lint.

**Comment by he-thinks** [comment]
> Shouldn't this be on the rust-lang/rust repo? `unused_parens` is a rustc lint, not a clippy lint.

I reopened this issue in the rust-lang/rust repo, thanks. [#117380](https://github.com/rust-lang/rust/issues/117380)
